overlord/assertstate/validation_set_tracking.go

```diff
@@ -172,3 +181,81 @@ func EnforcedValidationSets(st *state.State) (*snapasserts.ValidationSets, error
 
 	return sets, err
 }
+
+// AddCurrentTrackingToValidationSetsHistory stores the current state of validation-sets
+// tracking on top of the validation sets history.
+func AddCurrentTrackingToValidationSetsHistory(st *state.State) error {
+	all, err := ValidationSets(st)
+	if err != nil {
+		return err
+	}
+	return addToValidationSetsHistory(st, all)
+}
+
+func addToValidationSetsHistory(st *state.State, validationSets map[string]*ValidationSetTracking) error {
+	vshist, err := ValidationSetsHistory(st)
+	if err != nil {
+		return err
+	}
+
+	// if nothing is being tracked and history is empty (meaning nothing was
+	// tracked before, then don't store anything.
```

Comment:
I don't understand why it matters if history was there or not before

Reply:
My idea was to not store an empty validationSets map in the history as the only element of the history, because it's not useful to remember this.

However, if we were tracking something (history is not empty) and stop tracking it (validationSets is empty), then such empty set would be stored. See also the continuation of that comment.

Reply:
@bboozzoo In essence yes.  You are right in that it's most likely going to be called in places where either validationsSets is not empty, or vshist is not empty (on --forget). However I think it's better to have an explicit check here to ensure history is correct even if called with validationSets==nil.